### ISSUE TITLE
Undocumented differences between number and letter registers

### ISSUE THREAD

**Issue by SeerLite** [mawww/kakoune#4389]
## set-register
1. Open file with 3 empty lines and make cursor on each of them (e.g. with `2C`).
2. Set each register to a single value.
```kak
reg 1 hello
reg a hello
```

## -itersel
1. Open file with 3 empty lines and make a cursor on each of them (e.g. with `2C`).
2. Set register to 3 different values.
3. Paste all register contents (`<a-P>`) inside `execute-keys -itersel` block.

### Number register
```kak
:reg 1 one two three
:exec -itersel '"1' <a-P>
```
Result:
```plain
one
two
three
```
Number register iterates over all the registers along with the selections.

### Letter register
```kak
:reg a one two three
:exec -itersel '"a' <a-P>
```

Result:
```plain
onetwothree
onetwothree
onetwothree
```
All values of register `a` are used.

**Comment by SeerLite** [comment]
That is true. I actually realized the `set-register` difference because I was trying to figure out why in `exec/eval -itersel` captures worked how I expected while letter registers didn't.

Still, the number registers are assignable, unlike other registers that make `set-register` fail with an error message. If they're gonna be kept assignable, it'd be nice to make their behavior clear and document them.

Otherwise, I'd actually like to have the option to iterate over the letter registers like how number ones currently do, in `-itersel`. For now I'm abusing the (undocumented) behavior of number registers to get what I want.

**Comment by Screwtapello** [comment]
I think the behaviour you want is the behaviour of `<P>` without `-itersel`. If you want to operate on a set of selections and a set of register values in parallel, that's Kakoune's default mode of operation. `-itersel` is an escape hatch for special occasions, and you're apparently looking for an escape hatch from the escape hatch rather than just going back to the original mode, and that seems hacky to me.
